src/common/AdaptiveTreeFactory.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ *  Copyright (C) 2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "AdaptiveTreeFactory.h"
+#include "AdaptiveUtils.h"
+
+#include "../utils/PropertiesUtils.h"
+#include "../utils/FileUtils.h"
+#include "../utils/StringUtils.h"
+#include "../utils/log.h"
+#include "../parser/DASHTree.h"
+#include "../parser/HLSTree.h"
+#include "../parser/SmoothTree.h"
+
+using namespace PLAYLIST;
+using namespace UTILS;
+
+namespace
+{
+
+PROPERTIES::ManifestType InferManifestType(std::string_view url,
```

Comment:
Just a comment for later - we should introduce some tests for this.

Reply:
i will try take a look to include it here

Reply:
tests added

Reply:
Awesome, thanks!